activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -746,10 +747,20 @@ def pk_and_sequence_for(table)
         end
       end
 
-      # Returns just a table's primary key
-      def primary_key(table)
```

Comment:
`primary_key` is still public API. It is defined in `SchemaStatements#primary_key`.
[lib/active_record/connection_adapters/abstract/schema_statements.rb#L96-L101](https://github.com/kamipo/rails/blob/correctly_dump_composite_primary_key/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L96-L101)

```ruby
      # Returns just a table's primary key
      def primary_key(table_name)
        pks = primary_keys(table_name)
        return nil unless pks.size == 1
        pks.first
      end
```